src/platform/utilities/data/omit.js

```diff
@@ -5,11 +5,17 @@
  * @param {Object} object
  */
 export default function omit(fields, object) {
-  return Object.keys(object).reduce((newObj, k) => {
+  let fieldOmitted = false;
```

Comment:
Found a bug in `omit` where if nothing was actually omitted, we'd still get a different reference to the root object. I discovered it because this was making a test for `removeDeeplyEmptyObjects` fail.

Reply:
This makes sense to me, but I'm not sure what the expected behavior is. If you're using `omit` from the data utils, do you expect to always get a new object? Or do you only expect to get a new object when a field is actually omitted? Maybe a silly question but I don't know the answer.

Reply:
I think in general, it should only return a new reference if something changed. The idea is so we can compare equality like `props.foo === newProps.foo` when `newProps.foo` is run through `omit()`.

Reply:
this is also a bug in `set()`, I believe, but I bypassed it in `removeDeeplyEmptyObjects` by just checking for equality first.

Reply:
Ahh right, right. That explanation makes sense, thanks!